src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+
+    `replication.synchro_queue_max_size` is measured in number of bytes to
+    be written (0 means unlimited, which was the default behaviour before).
+    This option affects only the behavior of the master, and defaults to
+    16 megabytes.
+
+    Now that `replication.synchro_queue_max_size` is set on the master node,
+    tarantool will discard new transactions that try to queue after the limit
+    is reached. If a transaction had to be discarded, user will get an error
+    message "The synchronous transaction queue is full".
+
+    This limitation does not apply during the recovery process.
+]])
+
+M['replication.synchro_quorum'] = format_text([[
+    A number of replicas that should confirm the receipt of a synchronous
+    transaction before it can finish its commit.
+
+    This option supports dynamic evaluation of the quorum number. For example,
+    the default value is `N / 2 + 1` where `N` is the current number of replicas
+    registered in a cluster. Once any replicas are added or removed, the
+    expression is re-evaluated automatically.
+
+    Note that the default value (`at least 50% of the cluster size + 1`)
+    guarantees data reliability. Using a value less than the canonical one
+    might lead to unexpected results, including a split-brain.
+
+    `replication.synchro_quorum` is not used on replicas. If the master fails,
+    the pending synchronous transactions will be kept waiting on the replicas
+    until a new master is elected.
+]])
+
+M['replication.synchro_timeout'] = format_text([[
+    For synchronous replication only. Specify how many seconds to wait for a
+    synchronous transaction quorum replication until it is declared failed and
+    is rolled back.
+
+    It is not used on replicas, so if the master fails, the pending synchronous
+    transactions will be kept waiting on the replicas until a new master is
+    elected.
+]])
+
+M['replication.threads'] = format_text([[
+    The number of threads spawned to decode the incoming replication data.
+
+    In most cases, one thread is enough for all incoming data. Possible values
+    range from 1 to 1000. If there are multiple replication threads, connections
+    to serve are distributed evenly between the threads.
+]])
+
+M['replication.timeout'] = format_text([[
+    A time interval (in seconds) used by a master to send heartbeat requests to
+    a replica when there are no updates to send to this replica. For each
+    request, a replica should return a heartbeat acknowledgment.
+
+    If a master or replica gets no heartbeat message for
+    `4 * replication.timeout` seconds, a connection is dropped and a replica
+    tries to reconnect to the master.
+]])
+
+-- }}} replication configuration
+
+-- {{{ roles configuration
+
+M['roles'] = format_text([[
+    Specify the roles of an instance. To specify a role's configuration,
+    use the roles_cfg option.
+]])
+
+M['roles.*'] = 'Role name.'
```

Comment:
We can tell a bit more here. For example, that a role name is module name passed to the `require` call to load the role.

Reply:
New version:
```lua
I['roles.*'] = format_text([[
    The name of a role, corresponding to the module name used in the
    `require` call to load the role.
]])
```